Classes/EventStore/EventStream.php

```diff
@@ -51,13 +44,10 @@ public function __construct(\Iterator $streamIterator)
      */
     public function current()
     {
-        $configuration = new AllowAllPropertiesPropertyMappingConfiguration();
-
         /** @var RawEvent $rawEvent */
         $rawEvent = $this->streamIterator->current();
-        $eventClassName = $this->eventTypeResolver->getEventClassNameByType($rawEvent->getType());
         return new EventAndRawEvent(
-            $this->propertyMapper->convert($rawEvent->getPayload(), $eventClassName, $configuration),
+            $this->eventNormalizer->denormalize($rawEvent->getPayload(), $rawEvent->getType()),
```

Comment:
Why do we no longer need to resolve the eventType to a className?

Reply:
That's done in the `EventNormalizer` (which I'd like to rename to `EventConverter`) above, which will allow for centralized upcasting of events